src/core/ext/transport/chttp2/transport/hpack_parser.cc

```diff
@@ -1541,9 +1543,9 @@ static bool is_binary_literal_header(grpc_chttp2_hpack_parser* p) {
 static void set_precomputed_md_idx(grpc_chttp2_hpack_parser* p,
                                    grpc_mdelem md) {
   GPR_DEBUG_ASSERT(p->md_for_index.payload == 0);
-  GPR_DEBUG_ASSERT(p->precomputed_md_index == -1);
   p->md_for_index = md;
 #ifndef NDEBUG
+  GPR_DEBUG_ASSERT(p->precomputed_md_index == -1);
```

Comment:
Same here.

Reply:
Same response here also, and same offer.

Reply:
I'll revert this change also.